lib/rpmte.cc

```diff
@@ -159,9 +159,18 @@ static int addTE(rpmte p, Header h, fnpyKey key, rpmRelocation * relocs)
     p->arch = headerGetAsString(h, RPMTAG_ARCH);
     p->os = headerGetAsString(h, RPMTAG_OS);
 
-    /* gpg-pubkey's dont have os or arch (sigh), for others they are required */
-    if (!rstreq(p->name, "gpg-pubkey") && (p->arch == NULL || p->os == NULL))
+    if (p->arch == NULL || p->os == NULL) {
+	if (p->type == TR_REMOVED && rstreq(p->name, "gpg-pubkey")) {
+	    rpmlog(RPMLOG_WARNING, "erasing gpg-pubkey packages is deprecated. use rpmkeys --delete\n");
```

Comment:
If you use a period then you'd need to continue the next sentence with a capital letter. Which then requires that for the first sentence too. Use a comma, maybe? And like I already suggested, we could just as easily print out the actual rpmkeys command to use for deleting this particular key, so I think we should.